src/main/java/frc/robot/subsystems/Maflipulator.java

```diff
@@ -0,0 +1,176 @@
+            }
+
+            @Override
+            protected boolean isFinished() {
+                return false;
+            }
+        };
+    }
+
+    protected double restrict(double flipSpeed) {
+        if (currentPosition == MaflipulatorSide.kFront) {
+            if (flipSpeed > 0 && anglePot.get() >= FRONT_UPPER_ANGLE) {
+                flipSpeed = 0;
+            }
+            if (flipSpeed < 0 && anglePot.get() <= FRONT_LOWER_ANGLE) {
+                flipSpeed = 0;
+            }
+        } else {
+            if (flipSpeed > 0 && anglePot.get() <= BACK_UPPER_ANGLE) {
+                flipSpeed = 0;
+            }
+            if (flipSpeed < 0 && anglePot.get() >= BACK_LOWER_ANGLE) {
+                flipSpeed = 0;
+            }
+        }
+        return flipSpeed;
+    }
+
+    public Command restrictRotate() {
+        // The command is named "Restrict Rotate" and requires this subsystem.
+        return new Command("Restrict Rotate", this) {
+
+            @Override
+            protected void execute() {
+                double flipSpeed = Robot.xBoxCoPilot.getY(Hand.kRight);
+                flipSpeed = restrict(flipSpeed);
+                flipMotor.set(flipSpeed);
+            }
+
+            @Override
+            protected boolean isFinished() {
+                return false;
+            }
+
+            @Override
+            protected void end() {
+                // Called once after isFinished returns true
+            }
+        };
+    }
+
+    public Command Flip() {
+        // The command is named "Flip" and requires this subsystem.
+        return new Command("Flip", this) {
+            @Override
+            protected void initialize() {
+                if (currentPosition == MaflipulatorSide.kFront) {
+                    flipMotor.set(FLIP_MOTOR_SPEED);
+                } else {
+                    flipMotor.set(-FLIP_MOTOR_SPEED);
+                }
+            }
+
+            @Override
+            protected void execute() {
+                // Called repeatedly when this Command is scheduled to run
+            }
+
+            @Override
+            protected boolean isFinished() {
+                // Make this return true when this Command no longer needs to run execute()
+                if (currentPosition == MaflipulatorSide.kFront && anglePot.get() >= 270) {
+                    return true;
+                } else if (anglePot.get() <= 90) {
+                    return true;
+                }
+                return false;
+            }
+
+            @Override
+            protected void end() {
+                flipMotor.set(0);
+                // Called once after isFinished returns true
+            }
+        };
+    }
+
+    public Command moveToPosition() {
+        return new Command("move to position", this) {
+
+            @Override
+            protected void initialize() {
+                anglePID.reset();
+                anglePID.setSetpoint(0);
```

Comment:
Don't we need to be able to set the setpoint to a value, not just hardcode it to 0?